org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/conformance/TypeConformanceComputer.java

```diff
@@ -252,6 +253,36 @@ protected LightweightTypeReference getCommonSuperType(List<LightweightTypeRefere
 			if (allPrimitiveVoid(types)) {
 				return types.get(0);
 			}
+			
+			// Compute the common super type without looking at any of the UnboundTypeReferences that do
+			// not carry any hints yet.
+			List<LightweightTypeReference> unconstrainedUnboundTypes = null;
+			for(LightweightTypeReference type: types) {
```

Comment:
Maybe just nitpicking but why not create the array list here and in 271 check whether it's empty?
This could simplify the rest of the code?

Reply:
I was thinking that it's better to avoid the allocation of the array list, but given that recent Java versions don't create the inner array immediately, it can be simplified, true.

Reply:
Adjusted.